### ISSUE TITLE
OpenFAST build errors on macOS missing MPI symbols when building as shared library

### ISSUE THREAD

**Issue by jrood-nrel** [OpenFAST/openfast#208]
I'm seeing these errors during the 99% linking phase when building on macOS with OpenMPI 3.1.3:
```
Undefined symbols for architecture x86_64:
  "_MPI_Allreduce", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
  "_MPI_Comm_create", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_MPI_Comm_free", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
  "_MPI_Comm_group", referenced from:
      fast::OpenFAST::setInputs(fast::fastInputs const&) in OpenFAST.cpp.o
  "_MPI_Comm_rank", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
      fast::OpenFAST::setInputs(fast::fastInputs const&) in OpenFAST.cpp.o
  "_MPI_Comm_size", referenced from:
      fast::OpenFAST::allocateTurbinesToProcsSimple()     in OpenFAST.cpp.o
  "_MPI_Group_free", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
  "_MPI_Group_incl", referenced from:
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_ompi_mpi_comm_null", referenced from:
      fast::OpenFAST::end()    in OpenFAST.cpp.o
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
      fast::OpenFAST::allocateMemory()     in OpenFAST.cpp.o
  "_ompi_mpi_double", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
  "_ompi_mpi_op_sum", referenced from:
      fast::OpenFAST::fillScInputsGlob()     in OpenFAST.cpp.o
ld: symbol(s) not found for architecture x86_64
```
I will update this when I test whether or not OpenMPI 1.10.4 still works, which was previously known to work.

**Comment by jrood-nrel** [comment]
Same errors on macOS with OpenMPI 1.10.4 which worked previously before I had to temporarily shut off Nalu testing on the Mac.

**Comment by jrood-nrel** [comment]
With the help of @gantech I was able to build this now. I just have to figure out what I was doing wrong previously.

**Comment by jrood-nrel** [comment]
@gantech I found the difference in what I was doing originally that causes this problem. If I add `-DBUILD_SHARED_LIBS:BOOL=ON` to build OpenFAST as a shared library, that's what is causing my issue. It was building statically by default and I just realized this.

**Comment by jrood-nrel** [comment]
More specifically it's clearly missing linking to `libmpi` on the link line when I build as a shared library where `libmpi` is on the link line when I build as static.

**Comment by gantech** [comment]
Thanks @jrood-nrel. I'll try to add the link to `libmpi` for building the library as well.

**Comment by rafmudaf** [comment]
fixed by #242
